src/plugins/explore/public/application/utils/state_management/actions/query_actions.ts

```diff
@@ -56,6 +64,25 @@ export const defaultResultsProcessor = (rawResults: any, indexPattern: any) => {
   return result;
 };
 
+export const histogramResultsProcessor = (
+  rawResults: ISearchResult,
+  indexPattern: IndexPattern,
+  data: DataPublicPluginStart,
+  interval: string
+) => {
+  const result = defaultResultsProcessor(rawResults, indexPattern);
+  const histogramConfigs = createHistogramConfigs(indexPattern, interval, data);
+  const bucketAggConfig = histogramConfigs!.aggs[1] as IBucketDateHistogramAggConfig;
+  const tabifiedData = search.tabifyAggResponse(histogramConfigs!, rawResults);
+  const dimensions = getDimensions(histogramConfigs!, data);
+
+  result.bucketInterval = bucketAggConfig.buckets?.getInterval();
+  // @ts-ignore tabifiedData is compatible but due to the way it is typed typescript complains
+  result.chartData = buildPointSeriesData(tabifiedData, dimensions);
```

Comment:
Can we not using non-null assertion here? what about this
```suggestion
  const histogramConfigs = createHistogramConfigs(indexPattern, interval, data);
  if (histogramConfigs) {
    const bucketAggConfig = histogramConfigs.aggs[1] as IBucketDateHistogramAggConfig;
    const tabifiedData = search.tabifyAggResponse(histogramConfigs, rawResults);
    const dimensions = getDimensions(histogramConfigs, data);

    result.bucketInterval = bucketAggConfig.buckets?.getInterval();
    // @ts-ignore tabifiedData is compatible but due to the way it is typed typescript complains
    result.chartData = buildPointSeriesData(tabifiedData, dimensions);
  }
```